json_serializable/lib/src/encoder_helper.dart

```diff
@@ -16,6 +16,26 @@ import 'unsupported_type_error.dart';
 abstract class EncodeHelper implements HelperCore {
   String _fieldAccess(FieldElement field) => '$_toJsonParamName.${field.name}';
 
+  /// Generates an object containing metadatas related to the encoding,
+  /// destined to be used by other code-generators.
+  String createJsonMeta(Set<FieldElement> accessibleFieldSet) {
+    assert(config.createJsonMeta);
+
+    final buffer =
+        StringBuffer('const _\$${element.name}JsonMeta = <String, String> {');
```

Comment:
Careful using `element.name` – if it's already private you'll end up with double `_` which makes analyzer sad.

I have a helper around here somewhere for this"